### ISSUE TITLE
[BI3] Question: Library or Codec Issue for Battle Animations? 

### ISSUE THREAD

**Issue by habandy** [M-HT/SR#74]
Hi there. 
First of all: Thank you for this project. I found this while looking for custom maps for Battle Isle **2**. 😄 
I dont have tried every game listed. (i'll try both X-COM games as well) currently i'm experimenting to give it a try playing Battle Isle 3. 

I've played the GOG-Version a long time and always had a problem with the Battle Animations. 
They are starting as the units are shooting and ending shortly after the explosions.  (what i would describe as "starts too late and ends too early")
However in a non-GOG-Version this isnt the case.
There you have a few seconds before and after the shooting/explosions. 

How i know that?
Today it is still possible to play Battle Isle 3 on 64bit Windows, even without functional installation. 
Its a bit complicated and tricky, but there is a german guide for that online at [gamedoctorpc.de](https://www.gamedoctorpc.de/tutorials-spiele-games/spiele-games-b/battle-isle-3/)
But depending on your installation and hardware the battle animations can be flickering, half visible, **but are playing normally**.

I've the original BI3 CDs as well as the Battle Isle Platinum variant of the BI3 CDs. All german. 
The files from your project can be deployed and used as described. There are two steps needed to get it working: 
1) in the `BattleIsle3.cmd` file line 14 `set SDI_LANG=1` has to be changed to `set SDI_LANG=0`
2) in the `SDI.INI` file line 13 `COUNTRY=/1` has to be changed to `COUNTRY=/0`

Otherwise you are getting these errors because of not maching files/paths: 
1) `A severe error has occured! [...] MODULE: C:\DATA\WINCD\PC\SRC\W_MCP.C LINE: 1401`
2) `A severe error has occured! [...] MODULE: C:\DATA\WINCD\PC\SRC\W_MAIN.C LINE: 1299`

It can also be fixed by just overwriting the files in \ENG\ folder with the files from the GOG installation \DATA\ENG\ folder. 
But thats the dirty way and shouldnt be used, i guess. 

Back to my question / problem:
As soon as i place and use your project files i have the exact same battle animation issue with my non-gog-installation.
The battle animation is showing properly without flickering, but again starts too late and ended too early.
My question is: Does your project/solution differ from the GOG implementation ? 
My second question: Is this a Library/Codec issue? 
My third question: How could this be fixed? 

Let me know if you need more informations about that german installation.
I'm Using Windows 11 24H2 64bit.

**Comment by M-HT** [comment]
Battle Animations are not animations in the sense of playing some video file. Battle Animations are 3d rendered by the game, so there's no codec involved.
I never played the German version before, but you're right, it renders battle animations differently than English/French versions (with more time at the start and end of the battle).
My project is based on the executable from the English version, which is almost the same as the executable from GOG version (only one byte is different). That's why battle animations work the same as in GOG release.
I'll see if I can find a fix in my project, but it doesn't affect game play so it's a low priority issue.
When not using my project, you can copy the executable `SDI_1R.EXE` (maybe also .dll files`BASELIBR.DLL`, `MS_FIGTR.DLL`, `WC_FIGTR.DLL`) from the German version to the GOG version and that should fix this problem.

**Comment by habandy** [comment]
Wow! Actually i never thought this could depend on the version of the game. Thats a very good point! 
And your suggestion did indeed fix my problem to 100% 

Thank you very very much. This was actually very important to me. 
Is there something i can do for you? Can i provide informormations about the executables? do you need something else? 
I'm not very deep invested in that compile/re-compile thing, so i sadly cant provide active help in that way.

**Comment by M-HT** [comment]
> Is there something i can do for you? Can i provide informormations about the executables? do you need something else?

No, thank you (unless you're willing to analyze/debug the code).

BTW the line `COUNTRY=` in file `SDI.INI` is not used by the game AFAICT, so it doesn't need to be changed.
